pygmt/src/select.py

```diff
@@ -128,6 +131,38 @@ def select(data=None, outfile=None, **kwargs):
         and **+i** reverses the tests to pass record with *z* value NOT in the
         given range. Finally, if **+c** is not used then it is automatically
         incremented for each new ``z_subregion`` argument, starting with 2.
+    dist2pt : str
+        *pointfile*\|\ *lon*/*lat*\ **+d**\ *dist*.
+        Pass all records whose locations are within *dist* of any of the
+        points in the ASCII file *pointfile*. If *dist* is zero, the 3rd
+        column of *pointfile* must have each point's individual radius of
+        influence. If you only have a single point you can specify
+        *lon*/*lat* instead of *pointfile*. Distances are Cartesian and in
+        user units. Alternatively, if ``region`` and ``projection`` are used,
+        the geographic coordinates are projected to map coordinates (in
+        centimeters, inches, meters, or points, as determined by
+        :gmt-term:`PROJ_LENGTH_UNIT`) before Cartesian distances are compared
+        to *dist*.
+    dist2line : str
+        *linefile*\ **+d**\ *dist*\ [**+p**].
+        Pass all records whose locations are within *dist* of any of the line
+        segments in the ASCII multiple-segment file *linefile*. If *dist* is
+        zero, we will scan each sub-header in *linefile* for an embedded
+        **-D**\ *dist* setting that sets each line's individual distance value.
+        Distances are Cartesian and in user units. Alternatively, if ``region``
```

Comment:
I am unsure about the **-fg** option in the [upstream GMT documenation](https://docs.generic-mapping-tools.org/dev/gmtselect.html#l):
> Distances are Cartesian and in user units; specify -fg to indicate spherical distances append a distance unit (see [Units](https://docs.generic-mapping-tools.org/dev/gmtselect.html#units)).

Reply:
It sounds like you could set arc degree (d), arc minute (m), arc second (s) as units, but these geographical units will be projected to Cartesian units (e.g. meter) when doing the `select` operation?

Reply:
I can try to look at this later this day, but not sure if I find the time 🙁.

Reply:
It is very likely that I will not have the time to look at this in more detail today. So, @weiji14 you can merge this PR to have it within the v0.10.0 release, as it has two approvals already.

Reply:
Sounds good.